src/rpc/rawtransaction.cpp

```diff
@@ -177,10 +183,12 @@ UniValue getrawtransaction(const JSONRPCRequest& request)
                 throw JSONRPCError(RPC_MISC_ERROR, "Block not available");
             }
             errmsg = "No such transaction found in the provided block";
+        } else if (!g_txindex) {
+            errmsg = "No such mempool transaction. Use -txindex to enable blockchain transaction queries";
+        } else if (!f_txindex_ready) {
+            errmsg = "No such mempool transaction. Blockchain transactions are still in the process of being indexed";
```

Comment:
Yes, that's right, an RPC consumer should wait until the index is built. Using the `RPC_INVALID_ADDRESS_OR_KEY` error code for the `"No such mempool transaction. Use -txindex to enable blockchain transaction queries"` clause also seems odd to me.

Perhaps the `RPC_IN_WARMUP` makes sense here, or perhaps a new `RPC_INDEX_UNAVAILABLE` error code.

Reply:
A distinct error code may make sense, though no opinion on what. Further, I'm not sure we should let GetTransaction call into the g_txindex if !f_txindex_ready (as it may return data from a stale block, but its also just generally not so useful). Maybe GetTransaction/FindTx can check if we're in sync yet.